bugwarrior/services/linear.py

```diff
@@ -0,0 +1,181 @@
+        IDENTIFIER: {"type": "string", "label": "Linear Identifier"},
+        TEAM: {"type": "string", "label": "Project ID"},
+        CREATOR: {"type": "string", "label": "Issue Creator"},
+        ASSIGNEE: {"type": "string", "label": "Issue Assignee"},
+        CREATED_AT: {"type": "date", "label": "Issue Created"},
+        UPDATED_AT: {"type": "date", "label": "Issue Updated"},
+        CLOSED_AT: {"type": "date", "label": "Issue Closed"},
+    }
+
+    UNIQUE_KEY = (URL,)
+
+    def parse_date(self, input):
+        '''Parse a date, stripping microseconds'''
+        parsed = super(LinearIssue, self).parse_date(input)
+        if parsed:
+            parsed = parsed.replace(microsecond=0)
+        return parsed
+
+    def to_taskwarrior(self):
+        description = self.record.get("description")
+        created = self.parse_date(self.record.get("createdAt"))
+        modified = self.parse_date(self.record.get("updatedAt"))
+        closed = self.parse_date(self.record.get("completedAt"))
+
+        # Get a value, defaulting empty results to the given default.
+        def get(v, k, default=None):
+            r = v.get(k, default)
+            if not r:
+                return default
+            return r
+
+        return {
+            "project": re.sub(
+                r"[^a-zA-Z0-9]", "_", get(get(self.record, "project", {}), "name", "")
+            ).lower(),
+            "priority": self.config.default_priority,
+            "annotations": get(self.extra, "annotations", []),
+            "tags": self.get_tags(),
+            self.URL: self.record["url"],
+            self.TITLE: get(self.record, "title"),
+            self.DESCRIPTION: description,
+            self.STATE: get(get(self.record, "state", {}), "name"),
+            self.IDENTIFIER: get(self.record, "identifier"),
+            self.TEAM: get(get(self.record, "team", {}), "name"),
+            self.CREATOR: get(get(self.record, "creator", {}), "name"),
+            self.ASSIGNEE: get(get(self.record, "assignee", {}), "name"),
+            self.CREATED_AT: created,
+            self.UPDATED_AT: modified,
+            self.CLOSED_AT: closed,
+        }
+
+    def get_tags(self):
+        labels = [
+            label["name"] for label in self.record.get("labels", {}).get("nodes", [])
+        ]
+        return self.get_tags_from_labels(labels)
+
+    def get_default_description(self):
+        return self.build_default_description(
+            title=self.record.get("title"),
+            url=self.record.get("url"),
+            number=self.record.get("identifier"),
+            cls="task",
+        )
+
+
+class LinearService(Service, Client):
+    ISSUE_CLASS = LinearIssue
+    CONFIG_SCHEMA = LinearConfig
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.session = requests.Session()
+        self.session.headers.update(
+            {"Authorization": self.config.api_token, "Content-Type": "application/json"}
```

Comment:
This enables keyring support.
```suggestion
            {"Authorization": self.get_password("api_token"), "Content-Type": "application/json"}
```